### ISSUE TITLE
PS2 on iOS causes glitches

### ISSUE THREAD

**Issue by Iamthebestintheworld222** [libretro/RetroArch#16399]
When I played a ps2 game on my iPhone 14 Pro Max, I'm using capcom fighting evolution as an example, the font and control pad started to glitch out.
And eventually, everything went black
![image](https://github.com/libretro/RetroArch/assets/79031143/25fdda1c-6abc-4ad1-bd59-fc766cec36e0)![image](https://github.com/libretro/RetroArch/assets/79031143/8afae66d-93b9-4b5b-bc54-98775583eee4)![image](https://github.com/libretro/RetroArch/assets/79031143/617f9e5d-2320-44af-9761-07d0bce4414e)

**Comment by warmenhoven** [comment]
Yeah the play! core is still "experimental" and has a lot of rendering problems on iOS in particular that it doesn't have on other platforms. I have the beginnings of a fix but want to understand a bit more about what's causing the poor interaction.

**Comment by whaleboss** [comment]
> Yeah the play! core is still "experimental" and has a lot of rendering problems on iOS in particular that it doesn't have on other platforms. I have the beginnings of a fix but want to understand a bit more about what's causing the poor interaction.

What is the current status?

**Comment by warmenhoven** [comment]
No one has worked on this since [the governing body declared a land rush](https://appleinsider.com/articles/24/04/26/retro-gold-rush-these-emulators-are-coming-to-the-app-store-soon).

**Comment by Iamthebestintheworld222** [comment]
It's not for that, I'm using AltStore for jit enabling with ios17-jit-win

**Comment by warmenhoven** [comment]
I understand, just that this hasn't been the priority since App Store availability became a possibility.

I'd recommend you side load the standalone play! app on iOS so you get a baseline for what the experience on your phone will likely be.

I'm not planning on coming back to this until I've worked through most of the issues that are on my todo list under the category "App Store Fallout".

**Comment by Iamthebestintheworld222** [comment]
I would do that, but it would take up another app ID, and the standalone play emulator doesn't have a speed up feature like retroarch.

**Comment by Iamthebestintheworld222** [comment]
A bit of an update, glitched out text is still shown, OpenGL and vulkan do different things, and metal does the same as OpenGL. I'm also doing a challenge for parappa 2 since I played the game for many years on the ps4/ps5.

**Comment by whaleboss** [comment]
> A bit of an update, glitched out text is still shown, OpenGL and vulkan do different things, and metal does the same as OpenGL. I'm also doing a challenge for parappa 2 since I played the game for many years on the ps4/ps5.

Any updates on the progress?

**Comment by hizzlekizzle** [comment]
No change, no.

**Comment by warmenhoven** [comment]
The play! core should now be roughy as good as the upstream gl renderer.
